src/lib/expression/expression_utils.hpp

```diff
@@ -18,10 +18,17 @@ class TransactionContext;
 
 /**
  * Utility to check whether two vectors of Expressions are equal according to AbstractExpression::operator==()
+ * Note that this function also pays respect to the order of elements.
  */
 bool expressions_equal(const std::vector<std::shared_ptr<AbstractExpression>>& expressions_a,
                        const std::vector<std::shared_ptr<AbstractExpression>>& expressions_b);
 
+/**
+ * Utility to check whether a vector of distinct Expressions is a subset of another vector of distinct Expressions
+ */
+bool expressions_subset(const std::vector<std::shared_ptr<AbstractExpression>>& expressions_a,
+                        const std::vector<std::shared_ptr<AbstractExpression>>& expressions_b);
+
```

Comment:
I wrote this function and a corresponding test before using `ExpressionUnorderedSet`. 
Currently, it is unused. Shall I delete it or should it stay as a utility function?

Reply:
See my review comment. I suggest that you decide which set of helper methods would be the smallest set to cover all needs. If this one is unused, there is no point in keeping it.

Reply:
Function is removed.